src/api/test-controller/execution-context.js

```diff
@@ -1,18 +1,32 @@
 import { createContext } from 'vm';
 import Module from 'module';
-import { dirname } from 'path';
+import path from 'path';
 import exportableLib from '../exportable-lib';
 
-const OPTIONS_KEY = Symbol('options');
+const OPTIONS_KEY      = Symbol('options');
+const NODE_MODULES_DIR = 'node_modules';
+
+function getModuleBasePaths (currentPath) {
+    const root      = path.parse(currentPath).root;
+    const parentDir = path.dirname(currentPath);
+    const parentPaths = parentDir === root ? [] : getModuleBasePaths(parentDir);
```

Comment:
I think we can avoid recursion here. You can just split `parentDir` by `path.sep`, traverse all path segments and concat `node_modules` in a simple `for` loop.

Reply:
Splitting the dir path by separator leads to different behaviors on Windows and *nix (paths like '/home/username/work...' or 'C:\Work\...')